src/channel.rs

```diff
@@ -0,0 +1,75 @@
+//! Pin a Nix channel
+//!
+//! This should be preferred over pinning the equivaleng `nixpkgs` git branch.
+
+use crate::*;
+
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Eq, Hash)]
+pub struct Pin {
+    pub name: String,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Eq)]
+pub struct ChannelVersion {
+    pub url: url::Url,
+}
+
+impl diff::Diff for ChannelVersion {
+    fn diff(&self, other: &Self) -> Vec<diff::Difference> {
+        diff::d(&[diff::Difference::new("url", &self.url, &other.url)])
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Eq)]
+pub struct ChannelHash {
+    pub hash: String,
+}
+
+impl diff::Diff for ChannelHash {
+    fn diff(&self, other: &Self) -> Vec<diff::Difference> {
+        diff::d(&[diff::Difference::new("hash", &self.hash, &other.hash)])
+    }
+}
+
+#[async_trait::async_trait]
+impl Updatable for Pin {
+    type Version = ChannelVersion;
+    type Hashes = ChannelHash;
+
+    async fn update(&self, _old: Option<&ChannelVersion>) -> Result<ChannelVersion> {
+        /* We want to get from something like https://channels.nixos.org/nixos-21.11
+         * to https://releases.nixos.org/nixos/21.11/nixos-21.11.335807.df4f1f7cc3f/nixexprs.tar.xz
+         */
+        let mut url = build_client()?
+            .get(&format!("https://channels.nixos.org/{}", self.name))
```

Comment:
Lets just use `https://channels.nixos.org/<name>/nixexpr.tar.xz` from the beginning. That removes the needs for the `TODO error handling` below and is also compatible with situation in which there might not be a redirect.